File: backend/app/schemas/dataset.py
import uuid
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class DatasetResponse(BaseModel):
    id: uuid.UUID
    name: str
    organization_id: uuid.UUID
    source: str
    original_filename: Optional[str] = None
    created_by: str
    file_size_bytes: Optional[int] = None
    total_rows: int
    fraud_count: int
    avg_risk_score: float
    model_version: str
    threshold_version: str
    feature_schema_version: str
    processing_duration_ms: Optional[float] = None
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DatasetListResponse(BaseModel):
    items: List[DatasetResponse]
    total: int

File: frontend/src/services/dataset.ts
import { api } from "./api";
import { API_ENDPOINTS } from "./constants";

export type CollectionSource = "API" | "CSV" | string;
export type CollectionStatus = "Uploading" | "Processing" | "Completed" | "Failed" | "Archived" | string;

export interface CollectionItem {
  id: string;
  name: string;
  organization_id: string;
  source: CollectionSource;
  original_filename?: string;
  created_by: string;
  file_size_bytes?: number;
  total_rows: number;
  fraud_count: number;
  avg_risk_score: number;
  model_version: string;
  threshold_version: string;
  feature_schema_version: string;
  processing_duration_ms?: number;
  status: CollectionStatus;
  created_at: string;
  updated_at: string;
}

export interface CollectionListResponse {
  items: CollectionItem[];
  total: number;
}

export interface CollectionFilters {
  source?: string;
  status?: string;
  limit?: number;
}

export const collectionService = {
  async listCollections(params?: CollectionFilters): Promise<CollectionListResponse> {
    const response = await api.get<CollectionListResponse>(API_ENDPOINTS.COLLECTIONS, {
      params,
    });
    return response.data;
  },

  async getCollection(id: string): Promise<CollectionItem> {
    const response = await api.get<CollectionItem>(`${API_ENDPOINTS.COLLECTIONS}/${id}`);
    return response.data;
  },
};
